perllib/FixMyStreet/DB/Result/User.pm

```diff
@@ -295,6 +295,15 @@ sub permissions {
 sub has_permission_to {
     my ($self, $permission_type, $body_ids) = @_;
 
+    # Nobody, including superusers, can have a permission which isn't available
+    # in the current cobrand.
+    if (my $cobrand = $self->result_source->schema->cobrand) {
+        my $cobrand_perms = $cobrand->available_permissions;
+        my $flat_perms = { map { %$_ } values %$cobrand_perms };
+        my %available = map { $_ => 1 } keys %$flat_perms;
```

Comment:
These two lines could be `my %available = map { %$_ } values %$cobrand_perms;`

No need to put in a hashref and then read it back out again (the issue before your last commit was you just needed `keys map`, no ref in between), but also no need to  rewrite the values to be `1` if you're just checking a value exists :)